tests/aws/services/sns/test_sns_regions.py

```diff
@@ -0,0 +1,100 @@
+import json
+
+import pytest
+
+from localstack.testing.pytest import markers
+from localstack.utils.strings import short_uid
+from localstack.utils.sync import retry
+
+
+class TestSnsPartitions:
+    # We only have access to the AWS partition, not CHINA/US-GOV/etc
+    @markers.aws.manual_setup_required
+    @pytest.mark.parametrize("region,partition", [("us-east-2", "aws"), ("cn-north-1", "aws-cn")])
+    def test_topic_in_different_partitions(self, account_id, client_factory, region, partition):
```

Comment:
I think the `client_factory` was renamed `regional_client_factory`? In any case, see my other comment, it is generally better to use `aws_client_factory` directly, as it is callable with a similar interface, but provides typing. The only reason to go the other way is if the client should be dynamically determined by a string with the service name.